src/components/TopBar.tsx

```diff
@@ -0,0 +1,519 @@
+                    transition: 'transform 0.1s ease-out',
+                  }}
+                />
+              </Clickable>
+            ) : (
+              <div style={{ width: 'calc(1rem + 1rem + 1rem + 2px + 36px)' }} />
+            ) // padding (l+r) + margin (l+r) + icon size
+          }
+          <a style={{ ...pageTitleStyles, display: 'flex', alignItems: 'center' }} href={href || Nav.getLink('root')}>
+            {topBarLogo()}
+            <div>{title && <h1 style={{ fontSize: '1em', fontWeight: 500, padding: 0, margin: 0 }}>{title}</h1>}</div>
+          </a>
+          <div style={{ display: 'flex', flexGrow: 1 }}>{children}</div>
+          <AlertsIndicator
+            style={{
+              margin: '0 1rem 0 0.5rem',
+              color: isTerra() ? 'white' : colors.dark(),
+            }}
+          />
+          {openFirecloudModal && <PreferFirecloudModal onDismiss={() => setOpenFirecloudModal(false)} />}
+        </div>
+      </div>
+      <RequiredUpdateAlert />
+      <SkipNavTarget ref={mainRef} />
+    </div>
+  );
+};
+
+const PreferFirecloudModal = ({ onDismiss }) => {
+  const [emailAgreed, setEmailAgreed] = useState<boolean>(true);
+  const [reason, setReason] = useState<string>('');
+  const [submitting, setSubmitting] = useState<boolean>(false);
+
+  const user = useStore(userStore);
+  const {
+    profile: { firstName, lastName },
+  } = user;
+  const email = user.profile.contactEmail ?? user.terraUser.email ?? '';
+  const currUrl = window.location.href;
+
+  const returnToLegacyFC = _.flow(
+    withErrorReporting('Error opting out of Terra'),
+    Utils.withBusyState(setSubmitting)
+  )(async () => {
+    await Ajax().User.profile.preferLegacyFirecloud();
+    if (emailAgreed === true || reason.length !== 0) {
+      await Ajax().Support.createSupportRequest({
+        name: `${firstName} ${lastName}`,
+        email,
+        description: reason,
+        subject: 'Opt out of Terra',
+        type: 'survey',
+        attachmentToken: '',
+        emailAgreed,
+        currUrl,
+      });
+    }
+    onDismiss();
+    window.location.assign(getConfig().firecloudUrlRoot);
+  });
+
+  const reasonContainerId = useUniqueId();
+  return (
+    <Modal onDismiss={onDismiss} title='Return to classic FireCloud' okButton={returnToLegacyFC}>
+      Are you sure you would prefer the previous FireCloud interface?
+      <FormLabel htmlFor={reasonContainerId}>Please tell us why</FormLabel>
+      <TextArea
+        id={reasonContainerId}
+        style={{ height: 100, marginBottom: '0.5rem' }}
+        placeholder='Enter your reason'
+        value={reason}
+        onChange={setReason}
+      />
+      <LabeledCheckbox checked={emailAgreed} onChange={setEmailAgreed}>
+        <span style={{ marginLeft: '0.5rem' }}>You can follow up with me by email.</span>
+      </LabeledCheckbox>
+      {submitting && <SpinnerOverlay />}
+    </Modal>
+  );
+};
+
+export default TopBar;
```

Comment:
if you can squeeze in another improvement, please have this not be a default export anymore, just export TobBar as named export and update consumers.  We have been moving away from those because of mocking and module tooling headaches.

Reply:
Happy to: only reason I didn't before was just avoiding making to wide of changes.

Reply:
it would be great as a focused followup PR too.  Not a blocker on this PR.

Reply:
oooh, you got it in.... sweeeeeet!